AlphaWallet/Transfer/Controllers/TransactionConfigurator.swift

```diff
@@ -63,10 +63,15 @@ class TransactionConfigurator {
         self.session = session
         self.account = account
         self.transaction = transaction
-
+        let gasLimit: BigInt
+        if let txGasLimit = transaction.gasLimit, txGasLimit < GasLimitConfiguration.maxGasLimit {
+            gasLimit = txGasLimit
+        } else {
+            gasLimit = GasLimitConfiguration.maxGasLimit
+        }
         self.configuration = TransactionConfiguration(
             gasPrice: min(max(transaction.gasPrice ?? GasPriceConfiguration.defaultPrice, GasPriceConfiguration.minPrice), GasPriceConfiguration.maxPrice),
-            gasLimit: transaction.gasLimit ?? GasLimitConfiguration.maxGasLimit,
```

Comment:
This is optional, but I think you can simplify it with:

```
let gasLimit = [transaction.gasLimit, GasLimitConfiguration.maxGasLimit].min() ?? GasLimitConfiguration.maxGasLimit
```

The annoying `??` is because `min()` returns an optional to cater for when the collection is empty.

Reply:
Yup, already aware of that but your way is more elegant